ql/src/java/org/apache/hadoop/hive/ql/metadata/Hive.java

```diff
@@ -5215,7 +5215,7 @@ private static void moveAcidFiles(String deltaFileType, PathFilter pathFilter, F
               bucketDest.toUri().toString());
           try {
             fs.rename(bucketSrc, bucketDest);
-            if (newFiles != null) {
+            if (newFiles != null && !newFiles.contains(bucketDest)) {
```

Comment:
Have you checked why there is a duplicate path here ? And also use Set instead of List. List will become very expensive if the number of files is large.

Reply:
Actually, Task reattempt generates more than one temporary files and at the time of actual copying of data from temporary to table loc, the destination address is fed into this "newFiles" list multiple times.  As a result, there were multiple files in fileList in TXN_WRITE_NOTIFICAION_LOG. And this is problematic when we initiate distcp from src to tgt and it fails with duplicationFilesException.

Reply:
"Task reattempt generates more than one temporary files" ..this de-duplication should be done in file sink itself. In this case the second move/rename will fail. And that will cause data correctness issue.